File: app/src/main/java/com/example/pam_project/database/tasks/TaskMapper.java
package com.example.pam_project.database.tasks;

import com.example.pam_project.features.tasks.list.TaskInformation;
import com.example.pam_project.utils.constants.TaskStatus;

public class TaskMapper {

    public TaskInformation toModel(final TaskEntity entity) {
        return new TaskInformation(entity.id, entity.name, entity.description,
                entity.priority, TaskStatus.getStatus(entity.status));
    }
}

File: app/src/main/java/com/example/pam_project/database/tasks/TaskMapper.kt
package com.example.pam_project.database.tasks

import com.example.pam_project.features.tasks.list.TaskInformation
import com.example.pam_project.utils.constants.TaskStatus

class TaskMapper {
    fun toModel(entity: TaskEntity?): TaskInformation {
        return TaskInformation(
            id = entity!!.id,
            title = entity.name,
            description = entity.description,
            isUrgent = entity.priority,
            status = TaskStatus.getStatus(entity.status)
        )
    }
}